src/SharedStream.mo

```diff
@@ -123,56 +125,58 @@ module {
     var concurrentChunksCounter : Nat = 0;
     public func sendChunk() : async* {
       #ok : Nat;
-      #err : { #Paused; #Busy; #SendChunkError : Text; #StreamClosed };
+      #err : ChunkError or { #Paused; #Busy; #SendChunkError : Text };
```

Comment:
I think besides returning these errors from `sendChunk` we should also provide getters to the state that can check these error conditions without attempting to send a chunk.

For example `isPaused() : Bool` and `isBusy() : Bool` who check these conditions and return the answer. This will be useful because then the aggregator can make the results part of his stats/status. So if someone queries the aggregator for its status then it can return, as part of the whole status, whether its stream is busy or paused. 

Alternatively to the two functions above, we can also provide a function `status()` whose return value is a variant and #Busy and #Paused are cases. To be considered. Though I guess it could be #Busy and #Paused at the same time. So maybe status has to return a record and `busy` and `paused` are boolean fields in that record (among other fields).

Reply:
Done (separate getters)